### ISSUE TITLE
allow editing keybindings

### ISSUE THREAD

**Issue by ashkitten** [Yellow-Dog-Man/Resonite-Issues#23]
### Is your feature request related to a problem? Please describe.

i don't use a qwerty keyboard, and resonite does not seem to support remapping the key bindings whatsoever. this makes it unusable for me in desktop mode.

### Describe the solution you'd like

add a keybindings menu

### Describe alternatives you've considered

🤷‍♀️

**Comment by LualtOfficial** [comment]
I also don't use qwerty and would love to see this maybe in the meanwhile someone could make a mod for it.

**Comment by Bisclaveret** [comment]
Would presets for Azerty or Dvorak be a stopgap?

**Comment by ashkitten** [comment]
> Would presets for Azerty or Dvorak work as a stopgap?

not really. i use a split keyboard and space is on the right half. i usually map backspace instead.

**Comment by LualtOfficial** [comment]
> Would presets for Azerty or Dvorak work as a stopgap?

This also wouldn't work for me since I use halmak, and before that a custom layout. (I also use a split myself)
I might look into making a mod for it in the meanwhile.

**Comment by Scary-Guy** [comment]
So I prefer to use the ESDF keys instead of WASD, as this frees up my pinky to use the 1, Q, A, and Z keys more easily.  Except I also use the Workman layout, so my ESDF is actually RSHT.

Now, I have a programmable keyboard that lets me set a custom layer.  So that is QWERTY but the first two top right rows are shifted over one (TQWER and GASDF respectively.) Which sort of works, but I have to switch layers to do that and type messages and I would really like to not have to do that.

**Comment by Stellanora64** [comment]
As mentioned in #125 please also allow the ability to completely disable some inputs. Namely secondary axis, which should in turn disable the locomotion override. (Just putting this here so it doesn't get lost in another closed issue, and was brought up again as some more new users believed their controllers weren't working because of this.)

**Comment by Frooxius** [comment]
We're unlikely to allow that, because that would result in you having broken controls. 

The locomotion override is a separate thing - you won't be able to disable that by removing the binding - the tool will still be expecting the input.

**Comment by Stellanora64** [comment]
Was I misinterpreting Shifty's response then? https://github.com/Yellow-Dog-Man/Resonite-Issues/issues/125#issuecomment-2073605682

**Comment by Dusty-Sprinkles** [comment]
Probably worth adding a "reset to defaults" button or buttons for each control, ideally I'd like to see options for even things like ui edit mode too but that's possibility just me

**Comment by shiftyscales** [comment]
I did say "could be" @Stellanora64. It was not confirmation of anything. But as I also highlighted there in that same comment, if we did go that route- we'd likely require the user to rebind it to something else. It is common in key rebinding menus to either warn the user about, or prevent missing binds.

**Comment by BlueTera** [comment]
Yeah it would be nice if there was an option to mute and ummute on controller without opening the dash.

**Comment by Casuallynoted** [comment]
> Yeah it would be nice if there was an option to mute and ummute on controller without opening the dash.

Is there currently a mod that allows for this when in VR? Was struggling with this a bit last night, and found this issue haha

**Comment by XDelta** [comment]
> Is there currently a mod that allows for this when in VR? Was struggling with this a bit last night, and found this issue haha

Not a mod solution nor a keybind but there was the addition of [Facet_Anchors](https://wiki.resonite.com/Facets#Facet_Anchors) which may assist in making it quicker to access in the meantime. Here is the video on what they are. https://www.youtube.com/watch?v=XmvXq9NaQKs
